notes/PRODUCT.md

```diff
@@ -31,6 +33,69 @@ What do we plan to achieve? What goals are we setting?
+Locals looking for things to do
+
+### Problem
+Often do I see myself looking up for things to do, not knowing know what I want or what's going on around. There are many specialized apps/sources for food, events etc. instead of a single source. It takes a lot of effort to go through each one of them and plan my day or weekend.
+
+In summary, there is a lack of a unified platform for local area recommendations that provides a unified user experience, personalized to the user's taste.
+
+### Vision
+One stop personalized recommendation engine for local events and things to do
+
+### Strategy
+How do we plan to build the solution, monetize etc. Vision and strategic thinking (UX) are a must for a successful product.
+Includes the following: 
+1. In what form will this be available to the user? An app, personal assistant via Alexa/robot etc.
+2. Sample approach: (top down): Defining Product vision, goals, defining roadmap in order.
+3. How will you onboard new users, how will you get access to various data sources/feeds etc.?
+4. When and how will you monetize? Will the app be free at first?
+The list goes on. However, this some parts of this are out of scope of this discussion.
+
+### Goals
+Coming up with product goals is the step that helps you translate your product strategy into an executable plan.
+1. Personalization: A recommendation engine personalized to your hobbies, interests, recent activities, spending patterns etc.
+2. Social Media: Leverage the power of social media platforms like Facebook, Twitter etc. to enhance your experience by sharing friends' local activities.
+3. Monetization: Either through ads, revenue sharing of the local deals, restaurant recommendations etc.
+4. Security: Means for authorization, authentication, secure access over https etc.
+5. Plan, organize my schedule: Ability to subscribe to events, plan my activies, manage my calendar, setup alerts/reminders etc.<br>
+...
+
+### Features
+Product features are specific means of achieving the goals you have set. A goal usually has many features, but a feature delivery may attribute success in achieving a multiple goals. 
+Here are some examples of features for "Social Media" goal.
+1. Show places recently visited or posted by friends on social platforms like Facebook, Twitter, Instagram.
+2. Ability to like/dislike a post in Facebook
+3. Ability to re-tweet a post you like 
+5. A timeline of events of your and your friends' activity over the past 2 weeks/month/year
+6. Ability to organize an event such as a dinner by inviting your social media friends
+
+## Mapping goals and features into agile methodology
+Having gone through the process of product thinking, laying out our goals and features, we can map them into epics, stories and sub-tasks in agile methodoly. Here is an example for the sample use case above: 
+
+### Epic
+While this is not a hard rule, an epic in this case eg. "Leverage Social Media to enahance user experience" can roughly be mapped into a goal.
+
```

Comment:
These sound like epics, whereas your sub-tasks are actually stories that need further decomposition into sub-tasks - we should agree on a concrete definitions of epic / story / subtask or slap a big disclaimer on the fact that there is no one-size-fits all definition (_disclaimer_ should only be used in expert mode and generally we should be erring on the side of classroom expectations i.e. what students are capable of executing)